src/mpid/ch4/netmod/ofi/ofi_send.h

```diff
@@ -356,6 +356,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_send(const void *buf, MPI_Aint count, MPI
         if (*request == NULL) {
             *request = MPIR_Request_create_complete(MPIR_REQUEST_KIND__SEND);
         } else {
+            MPIR_Request_add_ref(*request);
```

Comment:
This probably should be done at request creation time, meaning when workq code created it.

Reply:
Maybe. But I would prefer not to change that logic at this point. I have a plan to let workq request work similar to persistent request, i.e. we simply let netmod/shmmod always create new request, and attach that request to the workq as `->real_request`. It should make the code much cleaner and keep the `workq` code strictly above netmod/shmmod.

Reply:
I'll just note that always creating a new request in the persistent case completely misses the point of the persistent API. It's not a high-priority, but there is definitely room to improve the persistent implementation to be lighter weight.

Reply:
> I'll just note that always creating a new request in the persistent case completely misses the point of the persistent API.

I see. The point of persistent request is so we skip new request creation? That makes sense. To properly do it then, we'll need a separate path for persistent request. To start, at least we'll need netmod/shmmod API for send_init etc.

Reply:
Right, we would need those lower layers to be able to hook into the init and start functions. Then all the work of translating comm,rank->addr, datatype lookup, etc. could be cached in the request and re-used in a lightweight manner. Anyway, not relevant to this PR.

Reply:
Where is workq request created, for this `MPIDI_OFI_send` example?
Our downstream code at this point, the extra ref count is already added for handoff/workq case.

Reply:
> 
> 
> Where is workq request created, for this `MPIDI_OFI_send` example?
> Our downstream code at this point, the extra ref count is already added for handoff/workq case.

It's in `ch4_send.h`. I didn't touch that since it is scattered. I am thinking they could all be refactored into `MPIDI_workq_{pt2pt,csend,rma}_enqueue`.